pkg/cover/canonicalizer.go

```diff
@@ -87,12 +88,22 @@ func (can *Canonicalizer) NewInstance(modules []host.KernelModule) *Canonicalize
 	}
 }
 
-func (ci *CanonicalizerInstance) Canonicalize(cov []uint32) {
+func (ci *CanonicalizerInstance) Canonicalize(cov []uint32, sign signal.Serial) {
+	// Skip conversion if modules are not used.
+	if len(ci.moduleKeys) == 0 {
+		return
+	}
 	convertModulePCs(ci.moduleKeys, ci.instToCanonicalMap, cov)
+	convertSignals(ci.moduleKeys, ci.instToCanonicalMap, sign)
 }
 
-func (ci *CanonicalizerInstance) Decanonicalize(cov []uint32) {
+func (ci *CanonicalizerInstance) Decanonicalize(cov []uint32, sign signal.Serial) {
+	// Skip conversion if modules are not used.
+	if len(ci.canonical.moduleKeys) == 0 {
+		return
+	}
 	convertModulePCs(ci.canonical.moduleKeys, ci.canonicalToInstMap, cov)
```

Comment:
`convertModulePCs` and `convertSignals` are de facto doing the same job now, right?

So maybe introduce some helper `convertPC` function and transform ("map") both all PCs and all elements of Serial with that?

Reply:
That's a good point; done